x-pack/platform/plugins/shared/onechat/server/utils/a2a/create_agent_card.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AgentCard, AgentSkill } from '@a2a-js/sdk';
+import type { AgentDefinition } from '@kbn/onechat-common';
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { ToolsServiceStart } from '../../services/tools';
+
+interface CreateAgentCardParams {
+  agent: AgentDefinition;
+  baseUrl: string;
+  toolsService: ToolsServiceStart;
+  request: KibanaRequest;
+}
+
+export async function createAgentCard({
+  agent,
+  baseUrl,
+  toolsService,
+  request,
+}: CreateAgentCardParams): Promise<AgentCard> {
+  const registry = await toolsService.getRegistry({ request });
+  const availableTools = await registry.list({});
+
+  // Create skills based on agent's tools
+  const skills: AgentSkill[] = [];
+
+  const agentToolIds = new Set(
+    agent.configuration.tools.flatMap((selection) => selection.tool_ids || [])
+  );
+
+  for (const tool of availableTools) {
```

Comment:
likely something we have to revisit. This is a good workaround for now but we should clarify good practices for skills.